cli/pcluster/config/validators.py

```diff
@@ -839,3 +839,41 @@ def intel_hpc_validator(param_key, param_value, pcluster_config):
         )
 
     return errors, warnings
+
+
+def custom_chef_cookbook_validator(param_key, param_value, pcluster_config):
```

Comment:
I think we can move this logic in the `url_validator` given that we're using it for pre_install, post_install and other parameters that can be stored in an S3 bucket.

Reply:
Then when adding a new validator, let's also add a test in the `tests/pcluster/config/test_validators.py` module.

We already have the `test_url_validator` so if you're going to merge your code on it you can simply extend it to cover your use cases.

Reply:
I did move the logic in `url_validator`, but I think that `custom_chef_cookbook` is the only field that accepts `s3://` and also depends on the cluster being able to read the object, so I added a conditional check on `custom_chef_cookbook`.

As for the test, I wanted to test a failure scenario where the cookbook is not in a bucket that the cluster can read (`head_object` fails, and the bucket is not in either `s3_read_resource` or `s3_read_write_resource`). Unfortunately, I couldn't figure out how to make `MockedBoto3Request` raise a `ClientError`. I left the test in the commit, but commented out, in case you can help me with this.

Reply:
Thanks for addressing the comments, I added some new minor ones.

There are other parameters accepting `s3://` urls (i.e. `pre_install` and `post_install`) so I think your improvement can be generalized.

To raise an error in the `MockedBoto3Request` you can pass `generate_error=True`, see example [here](https://github.com/aws/aws-parallelcluster/blob/v2.7.0/cli/tests/pcluster/utils/test_pcluster_utils.py#L381).